src/main/kotlin/uk/gov/dluhc/printapi/service/CertificateDataRetentionService.kt

```diff
@@ -0,0 +1,42 @@
+package uk.gov.dluhc.printapi.service
+
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import uk.gov.dluhc.printapi.database.repository.CertificateRepository
+import uk.gov.dluhc.printapi.exception.CertificateNotFoundException
+import uk.gov.dluhc.printapi.mapper.SourceTypeMapper
+import uk.gov.dluhc.printapi.messaging.models.ApplicationRemovedMessage
+import javax.transaction.Transactional
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class CertificateDataRetentionService(
+    private val sourceTypeMapper: SourceTypeMapper,
+    private val certificateRepository: CertificateRepository,
+    private val certificateRemovalDateResolver: CertificateRemovalDateResolver
+) {
+    /**
+     * Sets the initialRetentionRemovalDate on a [uk.gov.dluhc.printapi.database.entity.Certificate], after the
+     * originating application is removed from the source system (e.g. VCA).
+     *
+     * @param message An [ApplicationRemovedMessage] sent from the source system.
+     */
+    @Transactional
+    fun handleSourceApplicationRemoved(message: ApplicationRemovedMessage) {
```

Comment:
Rightly or wrongly I think our normal pattern is that the service method will take a DTO rather than the SQS interface model type, which means your listener class above needs to map the model message into a DTO

Any thoughts on that approach ?

Reply:
I had the same thought, but followed the approach Squad A have already established in this repo 🔥 😉

Reply:
😮